controllers/appgroup_controller.go

```diff
@@ -71,11 +71,12 @@ func (r *ApplicationGroupReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	}
 	patch := client.MergeFrom(appGroup.DeepCopy())
 
-	statusHelper := helpers.StatusHelper{
-		Client:    r.Client,
-		Logger:    logr,
-		PatchFrom: patch,
-		Recorder:  r.Recorder,
+	statusHelper := &helpers.StatusHelper{
+		Client:                r.Client,
+		Logger:                logr,
+		PatchFrom:             patch,
+		Recorder:              r.Recorder,
+		WorkflowClientBuilder: r.WorkflowClientBuilder,
 	}
 	reconcileHelper := helpers.ReconcileHelper{
```

Comment:
`ReconcilerHelper` leverages the same fields available in `ApplicationGroupReconciler`. We can make this a member of `ApplicationGroupReconciler` similar to `StatusHelper`.

Reply:
Logger should be different each time since we add logging information to the logger around the application group object so I somewhat disagree with this comment because what is the cost of creating this every time? All the values (except for the patch) are pointers so there should be no memory expense

Reply:
No real expense - more from readability and DRY aspect - since a lot of the fields are made available in an object that is accessible across the method body.

Reply:
I see, idea was to use the scope things within the reconcile method so that they don't become polluted within the reconcile object itself, since parallel calls to reconcile shouldn't have polluted values that sit within the reconciler itself

Reply:
Sure - works for me this way too 👍!